src/dex/fluid-dex/fluid-dex-liquidity-proxy.ts

```diff
@@ -51,30 +54,21 @@ export class FluidDexEventPool extends StatefulEventSubscriber<FluidDexPoolState
    */
   async handleOperate(
     event: any,
-    state: DeepReadonly<FluidDexPoolState>,
+    state: DeepReadonly<FluidDexLiquidityProxyState>,
     log: Readonly<Log>,
-  ): Promise<DeepReadonly<FluidDexPoolState> | null> {
-    if (!(event.args.user in [this.pool])) {
-      return null;
-    }
+  ): Promise<DeepReadonly<FluidDexLiquidityProxyState> | null> {
     const resolverContract = new Contract(
       this.commonAddresses.resolver,
       ResolverABI,
       this.dexHelper.provider,
     );
-    const rawResult = await resolverContract.callStatic.getPoolReservesAdjusted(
-      this.pool,
-      {
-        blockTag: this.dexHelper.provider,
-      },
-    );
 
-    const generatedState = this.convertToFluidDexPoolState(rawResult);
+    const rawResult =
+      await resolverContract.callStatic.getAllPoolsReservesAdjusted({
+        blockTag: log.blockNumber,
+      });
 
-    this.setState(
-      generatedState,
-      await this.dexHelper.provider.getBlockNumber(),
-    );
+    const generatedState = this.convertToFluidDexPoolState(rawResult);
 
     return generatedState;
```

Comment:
Can we replace it to `return this.generateState(log.blockNumber)`?

Reply:
@KanievskyiDanylo updated